lib/src/rules/prefer_foreach.dart

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.prefer_foreach_call;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc = r'Use forEach call to only apply a function to all the elements.';
+
+const _details = r'''
+
+**DO** use forEach call if you are only going to apply a function or a method to all the elements of an iterable.
+
+**BAD:**
+```
+for (final key in map.keys.toList()) {
+  map.remove(key);
+}
+```
+
+**GOOD:**
+```
+map.keys.toList().forEach(map.remove);
+```
+
```

Comment:
Add a comment here saying that the rule is experimental and that code that the suggested change alters logic if users rely on side effects over the iterable, provide an example.